dotCMS/src/main/java/com/dotcms/graphql/CategoryToMapTransformer.java

```diff
@@ -0,0 +1,95 @@
+package com.dotcms.graphql;
+
+import com.dotcms.contenttype.model.field.CategoryField;
+import com.dotcms.contenttype.model.field.Field;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.categories.model.Category;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.liferay.portal.model.User;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+
+public class CategoryToMapTransformer {
+    final Map<String, Object> mapOfMaps;
+    final List<Category> cats ;
+
+
+    public CategoryToMapTransformer(final Contentlet con, final User user) {
+        if (con.getInode() == null) {
+            throw new DotStateException("Contentlet needs an inode to get fields");
+        }
+        try {
+            this.cats = APILocator.getCategoryAPI().getParents(con, user, true);
+        } catch (DotDataException | DotSecurityException e) {
+            throw new DotStateException(e);
+        }
+        final Map<String, Object> newMap = new HashMap<>();
+        if(cats!=null && !cats.isEmpty()) {
+
+            for (final Field field : con.getContentType().fields()) {
+                if (field instanceof CategoryField) {
+                    newMap.put(field.variable(), transform(field, con, user));
+                }
+            }
+
+        }
+
+        this.mapOfMaps = newMap;
+    }
+
+
+    public Map<String, Object> asMap() {
+        return this.mapOfMaps;
+    }
+
+
+
+    private Map<String, Object> transform(final Field field, final Contentlet con, final User user) {
+
+        final Map<String, Object> map = new HashMap<>();
+
+        try {
+
+            final Category parentCategory        = APILocator.getCategoryAPI().find(field.values(), user, true);
+            final List<Map<String, Object>> childCategories = new ArrayList<>();
+
+            if(parentCategory != null) {
+                for (Category category : cats) {
```

Comment:
this logic can be replaced by a lambda expression

Reply:
same as above